SwiftServer/Sources/HTTPRequest/HTTPRequestParser.swift

```diff
@@ -4,48 +4,52 @@ enum ParsingErrors: Error {
     case BadRequest
 }
 
-public class RequestLine {
-    public var httpMethod: String
-    public var target: String
-    public var httpVersion: String
-
-    init(requestLine: String) throws {
-        var requestLineArray = requestLine.components(separatedBy: " ")
-        if requestLineArray.count != 3 {
-            throw ParsingErrors.BadRequest
-        }
-        httpMethod = requestLineArray[0]
-        target = requestLineArray[1]
-        httpVersion = requestLineArray[2]
-    }
+public protocol HTTPRequestParse {
+    var startLine: RequestLineParse! { get }
+    var headers: String? { get }
+    var messageBody: String? { get }
+
 }
 
-public class HTTPRequestParser {
-    public var startLine: RequestLine!
+public class HTTPParsedRequest: HTTPRequestParse {
+    public var startLine: RequestLineParse!
     public var headers: String?
     public var messageBody: String?
 
     public init(request: String) throws {
-        try parseHTTPRequest(request: request)
+        let messageParts = try parseHTTPRequest(request: request)
+        messageBody = messageParts.messageBody ?? nil
+        let headerAndRequestLine = parseHeaderAndRequestLine(
+                headerAndMessageRequestLine: messageParts.headerAndRequestLine)
+        headers = headerAndRequestLine.headers ?? nil
+        startLine = try parseRequestLine(requestLine: headerAndRequestLine.requestLine)
+    }
+
+    func parseRequestLine(requestLine: String) throws -> RequestLineParse {
+        guard let requestLine = try? RequestLine(requestLine: requestLine) else {
+            throw ParsingErrors.BadRequest
```

Comment:
Do you need to throw the BadRequest error here, if you're also throwing it in the parser?